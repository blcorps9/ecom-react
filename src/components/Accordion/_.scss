.accordion {
  .card-header {
    @extend .d-flex;
    @extend .flex-row;
    @extend .justify-content-between;
    @extend .align-items-center;
  }
}
