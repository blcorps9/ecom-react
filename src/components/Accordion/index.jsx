import React, { Component } from "react";
import cx from "classnames";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export default class Accordion extends Component {
  state = {
    openKey: "",
  };

  toggleCell = (e) => {
    e.preventDefault();
    e.stopPropagation();

    const id = e.currentTarget.getAttribute("data-id");

    this.setState((pS) => {
      if (pS.openKey === id) {
        return { openKey: "" };
      }

      return { openKey: id };
    });
  };

  render() {
    const { openKey } = this.state;
    const cells = this.props.cells;

    return (
      <div className="accordion">
        {cells.map((cell) => {
          const isOpen = openKey === cell.id;

          return (
            <div
              className="card"
              key={cell.id}
              data-id={cell.id}
              onClick={this.toggleCell}
            >
              <div className="card-header p-1">
                <h2 className="mb-0">
                  <div className="btn btn-block text-left" type="button">
                    {cell.header}
                  </div>
                </h2>
                <FontAwesomeIcon
                  icon={{
                    prefix: "fas",
                    iconName: isOpen ? "chevron-up" : "chevron-down",
                  }}
                />
              </div>

              <div
                className={cx("collapse", {
                  show: isOpen,
                })}
              >
                <div className="card-body p-0">{cell.body}</div>
              </div>
            </div>
          );
        })}
      </div>
    );
  }
}
