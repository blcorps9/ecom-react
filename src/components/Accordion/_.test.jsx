import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";

import Accordion from "./index";

const CELLS = [
  { id: 1, header: "Header 1", body: "This is a cell body - 1" },
  { id: 2, header: "Header 2", body: "This is a cell body - 2" },
];

describe("<Accordion />", () => {
  it("renders successfully", () => {
    const component = renderer.create(<Accordion cells={CELLS} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    const { container } = render(<Accordion cells={CELLS} />);

    expect(container.firstChild.className).toBe("accordion");
  });
});
