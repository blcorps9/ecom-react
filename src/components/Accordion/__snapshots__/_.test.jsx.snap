// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders successfully 1`] = `
<div
  className="accordion"
>
  <div
    className="card"
    data-id={1}
    onClick={[Function]}
  >
    <div
      className="card-header p-1"
    >
      <h2
        className="mb-0"
      >
        <div
          className="btn btn-block text-left"
          type="button"
        >
          Header 1
        </div>
      </h2>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-down fa-w-14 "
        data-icon="chevron-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="collapse"
    >
      <div
        className="card-body p-0"
      >
        This is a cell body - 1
      </div>
    </div>
  </div>
  <div
    className="card"
    data-id={2}
    onClick={[Function]}
  >
    <div
      className="card-header p-1"
    >
      <h2
        className="mb-0"
      >
        <div
          className="btn btn-block text-left"
          type="button"
        >
          Header 2
        </div>
      </h2>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-down fa-w-14 "
        data-icon="chevron-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="collapse"
    >
      <div
        className="card-body p-0"
      >
        This is a cell body - 2
      </div>
    </div>
  </div>
</div>
`;
